tools/test.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Build and test the site content
+#
+# Requirement: html-proofer, jekyll
+#
+# Usage: See help information
+
+set -eu
+
+SITE_DIR="_site"
+
+_config="_config.yml"
+
+help() {
+  echo "Build and test the site content"
+  echo
+  echo "Usage:"
+  echo
+  echo "   bash ./tools/test.sh [options]"
+  echo
+  echo "Options:"
+  echo '     -c, --config   "<config_a[,config_b[...]]>"    Specify config file(s)'
+  echo "     -h, --help               Print this information."
+}
+
+build() {
+  # clean up
+  if [[ -d $SITE_DIR ]]; then
+    rm -rf "$SITE_DIR"
+  fi
+
+  # build
+  JEKYLL_ENV=production bundle exec jekyll \
+    build \
+    --destination "$SITE_DIR" \
+    --config "$_config"
+}
+
+test() {
+  bundle exec htmlproofer \
+    --disable-external \
+    --allow_hash_href \
+    "$SITE_DIR"
+}
+
+main() {
+  build
+  test
+}
```

Comment:
The test script still lacks baseurl processing. Please change it to the following:

```suggestion
main() {
  # clean up
  if [[ -d $SITE_DIR ]]; then
    rm -rf "$SITE_DIR"
  fi

  _baseurl="$(grep '^baseurl:' "$_config" | sed "s/.*: *//;s/['\"]//g;s/#.*//")"

  # build
  JEKYLL_ENV=production bundle exec jekyll build \
    --destination "$SITE_DIR$_baseurl" \
    --config "$_config"

  # test
  bundle exec htmlproofer "$SITE_DIR" \
    --disable-external \
    --check-html \
    --allow_hash_href
}
```